public/app/plugins/datasource/graphite/state/providers.ts

```diff
@@ -199,6 +215,11 @@ async function addAltTagSegments(
     return segment;
   });
 
+  if (tagSegments.length + altSegments.length >= MAX_SUGGESTIONS) {
+    tagSegments = tagSegments.slice(0, MAX_SUGGESTIONS / 2);
+    altSegments = altSegments.slice(0, MAX_SUGGESTIONS / 2);
+  }
```

Comment:
this means that for 9999 tagSegments and 2 altSegments this function returns 5002 items instead of 10000. is it intentional?

Reply:
Yep, it's just to ensure 10k is not exceeded. I think 5k is still quite a lot so I didn't want to convolute the code with something more sophisticated that would always get 10k.